.github/workflows/nodejs-release.yml

```diff
@@ -1,56 +1,76 @@
-name: CI
+name: nodejs-release
 env:
   DEBUG: napi:*
-  APP_NAME: js-glaredb
-  MACOSX_DEPLOYMENT_TARGET: '10.13'
+  APP_NAME: "glaredb"
+  MACOSX_DEPLOYMENT_TARGET: "10.13"
+  PROTOC: "${{ github.workspace }}/deps/protoc/bin/protoc"
 permissions:
   contents: write
   id-token: write
-'on':
+on:
   push:
-    branches:
-      - main
-    tags-ignore:
-      - '**'
-    paths-ignore:
-      - '**/*.md'
-      - LICENSE
-      - '**/*.gitignore'
-      - .editorconfig
-      - docs/**
-  pull_request: null
+    tags:
+      - "*"
+  workflow_dispatch:
+defaults:
+  run:
+    working-directory: bindings/nodejs
+concurrency:
+  group: nodejs-release-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   build:
     strategy:
-      fail-fast: false
+      fail-fast: true
       matrix:
         settings:
-          - host: macos-latest
+          - host: macos-12-xl
             target: x86_64-apple-darwin
             build: |
               yarn build
               strip -x *.node
-          - host: windows-latest
+          - host: windows-latest-8-cores
             build: yarn build
             target: x86_64-pc-windows-msvc
-          - host: ubuntu-latest
+          - host: ubuntu-2004-8-cores
             target: x86_64-unknown-linux-gnu
             docker: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-debian
             build: |-
-              set -e &&
+              set -eux &&
+              curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | bash -s -- --to . &&
+              ./just protoc && 
+              export PROTOC="$PWD/deps/protoc/bin/protoc" &&
+              cd bindings/nodejs &&
               yarn build --target x86_64-unknown-linux-gnu &&
               strip *.node
+          - host: macos-12-xl
+            target: aarch64-apple-darwin
+            build: |
+              rustup target add aarch64-apple-darwin
+              sudo rm -Rf /Library/Developer/CommandLineTools/SDKs/*;
+              export CC=$(xcrun -f clang);
+              export CXX=$(xcrun -f clang++);
+              SYSROOT=$(xcrun --sdk macosx --show-sdk-path);
+              export CFLAGS="-isysroot $SYSROOT -isystem $SYSROOT";
+              yarn build --target aarch64-apple-darwin
```

Comment:
I have an idea for a _follow up_ to use https://github.com/boneskull/midnight-smoker in the pipeline to verify the packages before we deploy them. I don't know how that interplays with the matrix.

We can use that to check the basics before we publish so that we know the tarball contains what it should and is executable. If we get fancy, we can assert the LICENSE or do further things, but I really think just having confidence that npm pack worked and can be downloaded and executed is nice.

Reply:
(I really do consider this a follow-up though, so we can create a ticket).

Example here: https://github.com/greyscaled/markdownlint-rule-title-case-style/blob/6cce481f3215c7dc94199896d33e6f98c2cecf65/.github/workflows/publish.yaml#L9-L32